main/src/cgeo/geocaching/utils/LogWriter.java

```diff
@@ -0,0 +1,71 @@
+package cgeo.geocaching.utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+
+public class LogWriter {
+    private PrintWriter logWriter = null;
+    private String prefix;
+
+    public LogWriter(final String prefix) {
+        this.prefix = prefix;
+        checkLogfile();
+    }
+
+    public void log(final String info) {
+        checkLogfile();
+        if (null != logWriter) {
+            logWriter.println("--------------------------------------------------------------------");
+            logWriter.println(CalendarUtils.formatDateTime("yyyy-MM-dd HH:mm:ss.SSS"));
+            logWriter.println(info);
+        }
+    }
+
+    public void d(final String info) {
+        Log.d(info);
+        log(info);
+    }
+
+    public void e(final String info) {
+        Log.e(info);
+        log(info);
+    }
+
+    public void i(final String info) {
+        Log.i(info);
+        log(info);
+    }
+
+    public void v(final String info) {
+        Log.v(info);
+        log(info);
+    }
+
+    public void w(final String info) {
+        Log.w(info);
+        log(info);
+    }
+
+    public void close() {
+        if (null != logWriter) {
+            log("end of logging");
+            logWriter.println("--------------------------------------------------------------------");
+            logWriter.close();
+            logWriter = null;
+        }
+    }
+
+    private void checkLogfile() {
+        if (null == logWriter && Log.isDebug()) {
```

Comment:
Since "checkLogfile" is called on every call to "LogWriter" again: if an exception is catched later (and logWriter stays null) then this check will be repeated again and again, and on every call to LogWriter it will again try to instantiate the logWriter. Maybe a flag should be used so trying to set up file once it failed is not repeated

Reply:
Doing the repeated call to 'checkLogfile()' is to catch if a user has switched on debug mode throughout the operation. I agree that this may be a rare case, as you can activate debug mode only using settings, which are reachable only if you close any other activity within c:geo, so this extra check can probably left out.

Reply:
Ok understood